### ISSUE TITLE
Linker failure: undefined reference to `Java_java_util_prefs_FileSystemPreferences_chmod'

### ISSUE THREAD

**Issue by cayhorstmann** [oracle/graal#2856]
**Describe the issue**

Creating a native image fails with the above-mentioned linker issue.

**Steps to reproduce the issue**

```
cd /tmp
mkdir test
cd test
curl -O http://horstmann.com/codecheck/server/codecheck.tar.gz
tar xf codecheck.tar.gz
native-image --no-fallback -jar /opt/codecheck/codecheck.jar
```
**Describe GraalVM and your environment:**

GraalVM graalvm-ce-java11-20.2.0
JDK 11.0.4
OS: Ubuntu Linux 20.04.1
Arch. amd64

**More details**
Consider adding the `--native-image-info` and  `--verbose` flags when building your native image and paste output below.

```
Executing [
/data/apps/graalvm-ce-java11-20.2.0/bin/java \
-XX:+UseParallelGC \
-XX:+UnlockExperimentalVMOptions \
-XX:+EnableJVMCI \
-Dtruffle.TrustAllTruffleRuntimeProviders=true \
-Dtruffle.TruffleRuntime=com.oracle.truffle.api.impl.DefaultTruffleRuntime \
-Dgraalvm.ForcePolyglotInvalid=true \
-Dgraalvm.locatorDisabled=true \
-Dsubstratevm.IgnoreGraalVersionCheck=true \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.aarch64=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.amd64=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code.site=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code.stack=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.common=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.aarch64=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.amd64=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.sparc=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.meta=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.runtime=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.services=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.sparc=ALL-UNNAMED \
--add-exports=org.graalvm.truffle/com.oracle.truffle.api=ALL-UNNAMED \
--add-opens=jdk.internal.vm.compiler/org.graalvm.compiler.debug=ALL-UNNAMED \
--add-opens=jdk.internal.vm.compiler/org.graalvm.compiler.nodes=ALL-UNNAMED \
--add-opens=jdk.unsupported/sun.reflect=ALL-UNNAMED \
--add-opens=java.base/jdk.internal.module=ALL-UNNAMED \
--add-opens=java.base/jdk.internal.ref=ALL-UNNAMED \
--add-opens=java.base/jdk.internal.reflect=ALL-UNNAMED \
--add-opens=java.base/java.io=ALL-UNNAMED \
--add-opens=java.base/java.lang=ALL-UNNAMED \
--add-opens=java.base/java.lang.reflect=ALL-UNNAMED \
--add-opens=java.base/java.lang.invoke=ALL-UNNAMED \
--add-opens=java.base/java.lang.ref=ALL-UNNAMED \
--add-opens=java.base/java.net=ALL-UNNAMED \
--add-opens=java.base/java.nio=ALL-UNNAMED \
--add-opens=java.base/java.nio.file=ALL-UNNAMED \
--add-opens=java.base/java.security=ALL-UNNAMED \
--add-opens=java.base/javax.crypto=ALL-UNNAMED \
--add-opens=java.base/java.util=ALL-UNNAMED \
--add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED \
--add-opens=java.base/sun.security.x509=ALL-UNNAMED \
--add-opens=java.base/jdk.internal.logger=ALL-UNNAMED \
--add-opens=org.graalvm.sdk/org.graalvm.nativeimage.impl=ALL-UNNAMED \
--add-opens=org.graalvm.sdk/org.graalvm.polyglot=ALL-UNNAMED \
--add-opens=org.graalvm.truffle/com.oracle.truffle.polyglot=ALL-UNNAMED \
--add-opens=org.graalvm.truffle/com.oracle.truffle.api.impl=ALL-UNNAMED \
-XX:+UseJVMCINativeLibrary \
-Xss10m \
-Xms1g \
-Xmx13150142464 \
-Duser.country=US \
-Duser.language=en \
-Djava.awt.headless=true \
-Dorg.graalvm.version=20.2.0 \
-Dorg.graalvm.config= \
-Dcom.oracle.graalvm.isaot=true \
-Djava.system.class.loader=com.oracle.svm.hosted.NativeImageSystemClassLoader \
-Xshare:off \
--module-path \
/data/apps/graalvm-ce-java11-20.2.0/lib/truffle/truffle-api.jar \
-javaagent:/data/apps/graalvm-ce-java11-20.2.0/lib/svm/builder/svm.jar \
-Djdk.internal.lambda.disableEagerInitialization=true \
-Djdk.internal.lambda.eagerlyInitialize=false \
-Djava.lang.invoke.InnerClassLambdaMetafactory.initializeLambdas=false \
-cp \
/data/apps/graalvm-ce-java11-20.2.0/lib/svm/builder/pointsto.jar:/data/apps/graalvm-ce-java11-20.2.0/lib/svm/builder/objectfile.jar:/data/apps/graalvm-ce-java11-20.2.0/lib/svm/builder/llvm-platform-specific-shadowed.jar:/data/apps/graalvm-ce-java11-20.2.0/lib/svm/builder/javacpp-shadowed.jar:/data/apps/graalvm-ce-java11-20.2.0/lib/svm/builder/svm-llvm.jar:/data/apps/graalvm-ce-java11-20.2.0/lib/svm/builder/llvm-wrapper-shadowed.jar:/data/apps/graalvm-ce-java11-20.2.0/lib/svm/builder/svm.jar \
'com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus' \
-watchpid \
686809 \
-imagecp \
/data/apps/graalvm-ce-java11-20.2.0/lib/svm/library-support.jar:/opt/codecheck/codecheck.jar \
-H:Path=/tmp/ni \
-H:+DumpTargetInfo \
-H:FallbackThreshold=0 \
-H:Class=com.horstmann.codecheck.Runner \
-H:Name=codecheck \
-H:CLibraryPath=/data/apps/graalvm-ce-java11-20.2.0/lib/svm/clibraries/linux-amd64 \

]
[codecheck:686828]    classlist:   1,744.89 ms,  0.96 GB
[codecheck:686828]        (cap):     866.34 ms,  0.96 GB
[codecheck:686828]        setup:   2,854.99 ms,  0.96 GB
# Building image for target platform: org.graalvm.nativeimage.Platform$LINUX_AMD64
# Using native toolchain:
#   Name: GNU project C and C++ compiler (gcc)
#   Vendor: linux
#   Version: 9.3.0
#   Target architecture: x86_64
#   Path: /usr/bin/gcc
# Using CLibrary: com.oracle.svm.core.posix.linux.libc.GLibC
[codecheck:686828]     (clinit):     325.15 ms,  2.30 GB
# Static libraries:
#   ../../data/apps/graalvm-ce-java11-20.2.0/lib/svm/clibraries/linux-amd64/liblibchelper.a
#   ../../data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libnet.a
#   ../../data/apps/graalvm-ce-java11-20.2.0/lib/svm/clibraries/linux-amd64/libffi.a
#   ../../data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libnio.a
#   ../../data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libjava.a
#   ../../data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libfdlibm.a
#   ../../data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libzip.a
#   ../../data/apps/graalvm-ce-java11-20.2.0/lib/svm/clibraries/linux-amd64/libjvm.a
# Other libraries: pthread,dl,z,rt
[codecheck:686828]   (typeflow):  13,491.21 ms,  2.30 GB
[codecheck:686828]    (objects):   8,864.29 ms,  2.30 GB
[codecheck:686828]   (features):     612.62 ms,  2.30 GB
[codecheck:686828]     analysis:  23,883.39 ms,  2.30 GB
[codecheck:686828]     universe:   1,131.84 ms,  2.30 GB
[codecheck:686828]      (parse):   5,424.34 ms,  2.30 GB
[codecheck:686828]     (inline):   3,310.14 ms,  2.21 GB
[codecheck:686828]    (compile):  28,322.07 ms,  3.36 GB
[codecheck:686828]      compile:  38,708.37 ms,  3.36 GB
[codecheck:686828]        image:   2,814.41 ms,  3.36 GB
[codecheck:686828]        write:     401.15 ms,  3.36 GB
Fatal error:java.lang.RuntimeException: java.lang.RuntimeException: There was an error linking the native image: Linker command exited with 1

Linker command executed:
/usr/bin/gcc -z noexecstack -Wl,--gc-sections -Wl,--dynamic-list -Wl,/tmp/SVM-9303061323304520844/exported_symbols.list -Wl,--exclude-libs,ALL -Wl,-x -o /tmp/ni/codecheck /tmp/SVM-9303061323304520844/codecheck.o /data/apps/graalvm-ce-java11-20.2.0/lib/svm/clibraries/linux-amd64/liblibchelper.a /data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libnet.a /data/apps/graalvm-ce-java11-20.2.0/lib/svm/clibraries/linux-amd64/libffi.a /data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libnio.a /data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libjava.a /data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libfdlibm.a /data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libzip.a /data/apps/graalvm-ce-java11-20.2.0/lib/svm/clibraries/linux-amd64/libjvm.a -v -L/tmp/SVM-9303061323304520844 -L/data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc -L/data/apps/graalvm-ce-java11-20.2.0/lib/svm/clibraries/linux-amd64 -lpthread -ldl -lz -lrt 

Linker command output:
Using built-in specs.
COLLECT_GCC=/usr/bin/gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/9/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none:hsa
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 9.3.0-10ubuntu2' --with-bugurl=file:///usr/share/doc/gcc-9/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,gm2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-9 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none,hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 9.3.0 (Ubuntu 9.3.0-10ubuntu2) 
COMPILER_PATH=/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/:/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/
LIBRARY_PATH=/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib/:/lib/x86_64-linux-gnu/:/lib/../lib/:/usr/lib/x86_64-linux-gnu/:/usr/lib/../lib/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../:/lib/:/usr/lib/
COLLECT_GCC_OPTIONS='-z' 'noexecstack' '-o' '/tmp/ni/codecheck' '-v' '-L/tmp/SVM-9303061323304520844' '-L/data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc' '-L/data/apps/graalvm-ce-java11-20.2.0/lib/svm/clibraries/linux-amd64' '-mtune=generic' '-march=x86-64'
 /usr/lib/gcc/x86_64-linux-gnu/9/collect2 -plugin /usr/lib/gcc/x86_64-linux-gnu/9/liblto_plugin.so -plugin-opt=/usr/lib/gcc/x86_64-linux-gnu/9/lto-wrapper -plugin-opt=-fresolution=/tmp/ccjvlmIi.res -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s -plugin-opt=-pass-through=-lc -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s --build-id --eh-frame-hdr -m elf_x86_64 --hash-style=gnu --as-needed -dynamic-linker /lib64/ld-linux-x86-64.so.2 -pie -z now -z relro -o /tmp/ni/codecheck -z noexecstack /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/Scrt1.o /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/crti.o /usr/lib/gcc/x86_64-linux-gnu/9/crtbeginS.o -L/tmp/SVM-9303061323304520844 -L/data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc -L/data/apps/graalvm-ce-java11-20.2.0/lib/svm/clibraries/linux-amd64 -L/usr/lib/gcc/x86_64-linux-gnu/9 -L/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu -L/usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib -L/lib/x86_64-linux-gnu -L/lib/../lib -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib -L/usr/lib/gcc/x86_64-linux-gnu/9/../../.. --gc-sections --dynamic-list /tmp/SVM-9303061323304520844/exported_symbols.list --exclude-libs ALL -x /tmp/SVM-9303061323304520844/codecheck.o /data/apps/graalvm-ce-java11-20.2.0/lib/svm/clibraries/linux-amd64/liblibchelper.a /data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libnet.a /data/apps/graalvm-ce-java11-20.2.0/lib/svm/clibraries/linux-amd64/libffi.a /data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libnio.a /data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libjava.a /data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libfdlibm.a /data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libzip.a /data/apps/graalvm-ce-java11-20.2.0/lib/svm/clibraries/linux-amd64/libjvm.a -lpthread -ldl -lz -lrt -lgcc --push-state --as-needed -lgcc_s --pop-state -lc -lgcc --push-state --as-needed -lgcc_s --pop-state /usr/lib/gcc/x86_64-linux-gnu/9/crtendS.o /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/crtn.o
/usr/bin/ld: /tmp/SVM-9303061323304520844/codecheck.o:(.data+0xbd8): undefined reference to `Java_java_util_prefs_FileSystemPreferences_chmod'
collect2: error: ld returned 1 exit status

	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at java.base/java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:600)
	at java.base/java.util.concurrent.ForkJoinTask.get(ForkJoinTask.java:1006)
	at com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:480)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:349)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:508)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:114)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus.main(NativeImageGeneratorRunner.java:537)
Caused by: java.lang.RuntimeException: There was an error linking the native image: Linker command exited with 1

Linker command executed:
/usr/bin/gcc -z noexecstack -Wl,--gc-sections -Wl,--dynamic-list -Wl,/tmp/SVM-9303061323304520844/exported_symbols.list -Wl,--exclude-libs,ALL -Wl,-x -o /tmp/ni/codecheck /tmp/SVM-9303061323304520844/codecheck.o /data/apps/graalvm-ce-java11-20.2.0/lib/svm/clibraries/linux-amd64/liblibchelper.a /data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libnet.a /data/apps/graalvm-ce-java11-20.2.0/lib/svm/clibraries/linux-amd64/libffi.a /data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libnio.a /data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libjava.a /data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libfdlibm.a /data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libzip.a /data/apps/graalvm-ce-java11-20.2.0/lib/svm/clibraries/linux-amd64/libjvm.a -v -L/tmp/SVM-9303061323304520844 -L/data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc -L/data/apps/graalvm-ce-java11-20.2.0/lib/svm/clibraries/linux-amd64 -lpthread -ldl -lz -lrt 

Linker command output:
Using built-in specs.
COLLECT_GCC=/usr/bin/gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/9/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none:hsa
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 9.3.0-10ubuntu2' --with-bugurl=file:///usr/share/doc/gcc-9/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,gm2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-9 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none,hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 9.3.0 (Ubuntu 9.3.0-10ubuntu2) 
COMPILER_PATH=/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/:/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/
LIBRARY_PATH=/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib/:/lib/x86_64-linux-gnu/:/lib/../lib/:/usr/lib/x86_64-linux-gnu/:/usr/lib/../lib/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../:/lib/:/usr/lib/
COLLECT_GCC_OPTIONS='-z' 'noexecstack' '-o' '/tmp/ni/codecheck' '-v' '-L/tmp/SVM-9303061323304520844' '-L/data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc' '-L/data/apps/graalvm-ce-java11-20.2.0/lib/svm/clibraries/linux-amd64' '-mtune=generic' '-march=x86-64'
 /usr/lib/gcc/x86_64-linux-gnu/9/collect2 -plugin /usr/lib/gcc/x86_64-linux-gnu/9/liblto_plugin.so -plugin-opt=/usr/lib/gcc/x86_64-linux-gnu/9/lto-wrapper -plugin-opt=-fresolution=/tmp/ccjvlmIi.res -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s -plugin-opt=-pass-through=-lc -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s --build-id --eh-frame-hdr -m elf_x86_64 --hash-style=gnu --as-needed -dynamic-linker /lib64/ld-linux-x86-64.so.2 -pie -z now -z relro -o /tmp/ni/codecheck -z noexecstack /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/Scrt1.o /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/crti.o /usr/lib/gcc/x86_64-linux-gnu/9/crtbeginS.o -L/tmp/SVM-9303061323304520844 -L/data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc -L/data/apps/graalvm-ce-java11-20.2.0/lib/svm/clibraries/linux-amd64 -L/usr/lib/gcc/x86_64-linux-gnu/9 -L/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu -L/usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib -L/lib/x86_64-linux-gnu -L/lib/../lib -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib -L/usr/lib/gcc/x86_64-linux-gnu/9/../../.. --gc-sections --dynamic-list /tmp/SVM-9303061323304520844/exported_symbols.list --exclude-libs ALL -x /tmp/SVM-9303061323304520844/codecheck.o /data/apps/graalvm-ce-java11-20.2.0/lib/svm/clibraries/linux-amd64/liblibchelper.a /data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libnet.a /data/apps/graalvm-ce-java11-20.2.0/lib/svm/clibraries/linux-amd64/libffi.a /data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libnio.a /data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libjava.a /data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libfdlibm.a /data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libzip.a /data/apps/graalvm-ce-java11-20.2.0/lib/svm/clibraries/linux-amd64/libjvm.a -lpthread -ldl -lz -lrt -lgcc --push-state --as-needed -lgcc_s --pop-state -lc -lgcc --push-state --as-needed -lgcc_s --pop-state /usr/lib/gcc/x86_64-linux-gnu/9/crtendS.o /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/crtn.o
/usr/bin/ld: /tmp/SVM-9303061323304520844/codecheck.o:(.data+0xbd8): undefined reference to `Java_java_util_prefs_FileSystemPreferences_chmod'
collect2: error: ld returned 1 exit status

	at com.oracle.svm.hosted.image.NativeBootImageViaCC.handleLinkerFailure(NativeBootImageViaCC.java:459)
	at com.oracle.svm.hosted.image.NativeBootImageViaCC.write(NativeBootImageViaCC.java:434)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:677)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$0(NativeImageGenerator.java:468)
	at java.base/java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1407)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Error: Image build request failed with exit status 1
com.oracle.svm.driver.NativeImage$NativeImageError: Image build request failed with exit status 1
	at com.oracle.svm.driver.NativeImage.showError(NativeImage.java:1558)
	at com.oracle.svm.driver.NativeImage.build(NativeImage.java:1308)
	at com.oracle.svm.driver.NativeImage.performBuild(NativeImage.java:1269)
	at com.oracle.svm.driver.NativeImage.main(NativeImage.java:1228)
	at com.oracle.svm.driver.NativeImage$JDK9Plus.main(NativeImage.java:1740)

```

**Comment by oubidar-Abderrahim** [comment]
Did you try to use the native-image agent to generate config files?

You can do so by using the command `java -agentlib:native-image-agent=config-output-dir=conf-dir -jar youApp.jar`
For generating the native-image you can use `native-image -H:ConfigurationFileDirectories=conf-dir -H:+ReportExceptionStackTraces -H:+StackTrace --no-server --no-fallback -jar youApp.jar`

**Comment by cayhorstmann** [comment]
Unfortunately, that makes no difference. It fails with the same linker error message.

**Comment by gradinac** [comment]
Hey @cayhorstmann!
The cause of this issue is that the `prefs` library is not being linked in properly - we have addressed this issue and the fix will be included in GraalVM 20.3 - you can test whether the fix would work using one of our nightly builds from https://github.com/graalvm/graalvm-ce-dev-builds/releases

You could also try adding the following option to the native-image builder to force it to link against the `prefs` library:
`-H:NativeLinkerOption=/data/apps/graalvm-ce-java11-20.2.0/lib/static/linux-amd64/glibc/libprefs.a`

**Comment by cayhorstmann** [comment]
Thanks very much for the fast turnaround. It works with the added option.
